### PR TITLE
Update spreadsheet validation errors

### PR DESCRIPTION
**Pull request by karlchillmaid**
This PR updates:
- some validation errors to make them clearer
- the button text

---

Screen Shot 2019-09-27 at 11 41 39

---

Screen Shot 2019-09-27 at 11 43 32

---

Screen Shot 2019-09-27 at 11 45 53

### CODE REVIEW

File: app/templates/views/check/column-errors.html [hunk 1]
```diff
@@ -56,49 +56,57 @@ <h1 class='banner-title' data-module="track-error" data-error-type="No rows" dat
...
         <p>
           Delete or rename one of these columns and try again.
         </p>
 
       {% elif recipients.missing_column_headers %}
 
         <h1 class='banner-title' data-module="track-error" data-error-type="Missing placeholder columns" data-error-label="{{ upload_id }}">
-          The columns in your file need to match the double brackets in
-          your template
+          Your file is missing some columns
```
Comment (quis) [inline app/templates/views/check/column-errors.html]: This is technically accurate, but less clear I think. Often the problem is that the columns are there, but mislabelled. So saying the columns are ‘missing’ might confuse people.

File: app/templates/views/check/column-errors.html [hunk 2]
```diff
@@ -141,7 +149,7 @@ <h1 class='banner-title' data-module="track-error" data-error-type="Missing plac
         {{ file_upload(
           form.file,
           action=url_for('.send_messages', service_id=current_service.id, template_id=template.id),
-          button_text='Re-upload your file'
+          button_text='Upload a new file'
```
Comment (quis) [inline app/templates/views/check/column-errors.html]: The behaviour here is often to fix the problem with the file (eg rename a column) and then upload it again. I don’t think ‘new file’ describes this well.

Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: Re-upload isn't a word, so we shouldn't use it.
My previous draft was 'Upload your file again'.
It shifted to this version after discussions with other content designers. There was a concern that encouraging users to upload again might not make it clear enough that they need to fix the file and save it again first.
Hopefully the wording of the error summary mitigates that risk, though.
Happy to go with 'Upload your file again'.

File: app/templates/views/check/column-errors.html [hunk 3]
```diff
@@ -56,49 +56,57 @@ <h1 class='banner-title' data-module="track-error" data-error-type="No rows" dat
...
 
         <h1 class='banner-title' data-module="track-error" data-error-type="Missing recipient columns" data-error-label="{{ upload_id }}">
-          Your file needs {{ (
-            recipients.missing_column_headers
-            if template.template_type == 'letter' else required_recipient_columns
-          ) | formatted_list(
-            prefix='a column called',
-            prefix_plural='columns called'
-          ) }}
+          Your file is missing some columns
```
Comment (quis) [inline app/templates/views/check/column-errors.html]: I think this has the same problem as the ‘double brackets‘ error message.

If the columns in someone’s file are:
```
email | name | date
```
and you say ‘Your file is missing some columns‘ their reaction will often be ‘no it’s not, all three columns are there‘. And this causes them to lose faith in the system’s ability to help them fix the problem.

An error message that says
> Your file needs a column called ‘email address’

Helps people understand that it’s about the naming of the column, not just that there is a column with some email addresses in it.

Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: What if they're missing multiple columns? How long can that first line of the error get, realistically?

Comment (quis) [reply app/templates/views/check/column-errors.html]: Longest example would be:
> Your file needs columns called ‘Address line 1’, ‘Address line 2’ and ’Postcode’

Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: Screen Shot 2019-09-26 at 14 36 55

*vs*

Screen Shot 2019-09-26 at 14 38 25

File: app/templates/views/check/column-errors.html (hunk 1 above)
Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: Screen Shot 2019-09-26 at 14 53 27

File: app/templates/views/check/column-errors.html (hunk 2 above)
Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: Screen Shot 2019-09-26 at 14 54 34

File: app/templates/views/check/column-errors.html (hunk 3 above)
Comment (karlchillmaid) [reply app/templates/views/check/column-errors.html]: Screen Shot 2019-09-26 at 15 17 49

Makes it a bit easier to compare when the column names are only _slightly_ wrong?

**Pull request merged**